rcon/commands.py

```diff
@@ -290,8 +348,22 @@ def get_map_rotation(self):
     def get_slots(self):
         return self._get("slots", can_fail=False)
 
-    def get_vip_ids(self):
-        return self._get("vipids", True, can_fail=False)
+    def get_vip_ids(self) -> List[VipId]:
+        with self.with_connection() as conn:
+            res = self._get("vipids", True, can_fail=False, conn=conn)
+
+            vip_ids: List[VipId] = []
+            for item in res:
+                try:
+                    steam_id_64, name = item.split(" ", 1)
+                    name = name.replace('"', "")
+                    name = name.replace("\n", "")
+                    name = name.strip()
+                    vip_ids.append(dict(steam_id_64=steam_id_64, name=name))
+                except ValueError:
+                    self._reconnect(conn)
+                    raise
+            return vip_ids
```

Comment:
Why the big change here? Isn't the extended_commands.Rcon doing that already?

Reply:
it did, however, it required the connection being exposed to handle the reconnect when reading the result of the vipids command. With this code being in extended_commands (while ServerCtl actually does the reequest to the server) there was not context for what connection was used to reconnect it. To keep the same functionality, I moved reading the list into here (assuming that no-one needs the raw list anyways, at least not in the current version of the code)